src/wp-includes/i18n/class-wp-translation-file.php

```diff
@@ -0,0 +1,302 @@
+<?php
+/**
+ * I18N: WP_Translation_File class.
+ *
+ * @package WordPress
+ * @subpackage I18N
+ * @since 6.5.0
+ */
+
+/**
+ * Class WP_Translation_File.
+ *
+ * @since 6.5.0
+ */
+abstract class WP_Translation_File {
+	/**
+	 * List of headers.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string, string>
+	 */
+	protected $headers = array();
+
+	/**
+	 * Whether file has been parsed.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var bool
+	 */
+	protected $parsed = false;
+
+	/**
+	 * Error information.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var bool|string
+	 */
+	protected $error = false;
+
+	/**
+	 * File name.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var string
+	 */
+	protected $file = '';
+
+	/**
+	 * Translation entries.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string, string>
+	 */
+	protected $entries = array();
+
+	/**
+	 * Plural forms function.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var callable|null Plural forms.
+	 */
+	protected $plural_forms = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $file File to load.
+	 */
+	protected function __construct( string $file ) {
+		$this->file = $file;
+	}
+
+	/**
+	 * Creates a new WP_Translation_File instance for a given file.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string      $file     File name.
+	 * @param string|null $filetype Optional. File type. Default inferred from file name.
+	 * @return false|WP_Translation_File
+	 */
+	public static function create( string $file, string $filetype = null ) {
+		if ( ! is_readable( $file ) ) {
+			return false;
+		}
+
+		if ( null === $filetype ) {
+			$pos = strrpos( $file, '.' );
+			if ( false !== $pos ) {
+				$filetype = substr( $file, $pos + 1 );
+			}
+		}
+
+		switch ( $filetype ) {
+			case 'mo':
+				return new WP_Translation_File_MO( $file );
+			case 'php':
+				return new WP_Translation_File_PHP( $file );
+			default:
+				return false;
+		}
+	}
+
+	/**
+	 * Returns all headers.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array<string, string> Headers.
+	 */
+	public function headers() {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+		return $this->headers;
+	}
+
+	/**
+	 * Returns all entries.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array<string, string[]> Entries.
+	 */
+	public function entries() {
+		if ( ! $this->parsed ) {
+			$this->parse_file();
+		}
+
+		return $this->entries;
+	}
+
+	/**
+	 * Returns the current error information.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool|string Error
```

Comment:
See above - also missing period :)

I would suggest:
```suggestion
	 * @return string Error text, or empty string if no error.
```